### PR TITLE
Upgrade tokenizer

### PR DESCRIPTION
**Pull request by sserdoubleh**

### CODE REVIEW

File: knover/core/model.py [hunk 1]
```diff
@@ -72,6 +72,8 @@ def add_cmdline_args(cls, parser):
         return group
 
     def __init__(self, args, place):
+        self.args = args
```
Comment (sserdoubleh) [inline knover/core/model.py:75]: Fix load checkpoints: load start_step from checkpoint.
https://github.com/PaddlePaddle/Knover/issues/59

File: knover/data/dialog_reader.py [hunk 2]
```diff
@@ -84,12 +84,12 @@ def __init__(self, args):
...
-        self.eos_id = args.eos_id = self.vocab["[SEP]"]
-        self.unk_id = args.unk_id = self.vocab["[UNK]"]
-        self.mask_id = args.mask_id = self.vocab["[MASK]"]
-        self.vocab_size = args.get("vocab_size", 0)
+        self.pad_id = args.pad_id = self.tokenizer.pad_id
+        self.bos_id = args.bos_id = self.tokenizer.bos_id
+        self.eos_id = args.eos_id = self.tokenizer.eos_id
+        self.unk_id = args.unk_id = self.tokenizer.unk_id
+        self.mask_id = args.mask_id = self.tokenizer.mask_id
+        self.vocab_size = args.get("vocab_size", self.tokenizer.vocab_size)
```
Comment (sserdoubleh) [inline knover/data/dialog_reader.py:92]: Using Tokenizer property.

File: knover/data/dialog_reader.py [hunk 3]
```diff
@@ -338,7 +338,7 @@ def _read_numerical_file(self, fp, phase, is_infer, delimiter=";"):
                 # if you change the numerical data format, you must to make sure the last part of
                 # numerical sequence is the target sequence
                 def rindex(lst, elem):
-                    return len(lst) - lst[-1:0:-1].index(elem) - 1
+                    return len(lst) - lst[::-1].index(elem) - 1
```
Comment (sserdoubleh) [inline knover/data/dialog_reader.py:341]: Get the last position of [BOS] token, which is the start position of text generation.
It is also suitable for autoregressive task.

**Review by portia1026 — APPROVED** [review]
LGTM

**Pull request merged**